cdap-api/src/main/java/co/cask/cdap/api/templates/ApplicationTemplate.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.templates;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.service.http.HttpServiceHandler;
+
+import javax.annotation.Nullable;
+
+/**
+ * Abstract App Template that provides additional functionality required for App Templates.
+ */
+@Beta
+public abstract class ApplicationTemplate extends AbstractApplication {
+
+  /**
+   * Given the manifest configuration, configures the Manifest.
+   *
+   * @param configuration manifest configuration
+   * @param configurer {@link ManifestConfigurer}
+   * @throws Exception if the configuration is not valid
+   */
+  public abstract void configureManifest(String configuration, ManifestConfigurer configurer) throws Exception;
```

Comment:
If it doesn't have anything to configure, we won't know what program to start. By having all this info in the configureManifest, we have removed the need to have any info in the MANIFEST file of etl jar.